pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java

```diff
@@ -212,7 +215,18 @@ public void postSchema(
                         .build()
                 ).build()
             )
-        );
+        ).exceptionally(error -> {
+            if (error instanceof IncompatibleSchemaException) {
+                response.resume(
+                    Response.status(Response.Status.BAD_REQUEST).build()
```

Comment:
Can we use something more specific than HTTP 400 error? That would help on the client side to figure out the reason of the error.